src/main/java/chatowo/Chatowo.java

```diff
@@ -0,0 +1,141 @@
+package chatowo;
+
+import chatowo.task.Task;
+import chatowo.task.TaskList;
+
+import java.io.IOException;
+
+/**
+ * Main class that handles the chat bot functionality.
+ * Manages tasks, user interactions, and file operations.
+ */
+public class Chatowo {
+    // initialise task list
+    private TaskList list = new TaskList();
+    private final Storage storage = new Storage("./data/chatowo.txt");
+    private final Ui ui = new Ui();
+    private final Parser parser = new Parser(this);
+
+    public void reply(String msg) {
+        ui.reply(msg);
+    }
+
+    public void listTasks() {
+        ui.listTasks(list);
+    }
+
+    public void dateTimeError() {
+        ui.dateTimeError();
+    }
```

Comment:
Should this be renamed to sound like a verb?
```suggestion
    public void reportDateTimeError() {
        ui.reportDateTimeError();
    }
```